### PR TITLE
Tab on user profile to show starred repos

### PR DESCRIPTION
**Pull request by andreynering**
Deprecates https://github.com/go-gitea/gitea/pull/181

Improvements pending for future PRs:

- Have a timestamp in star table and order by it
- Pagination
- Filter

### CODE REVIEW

File: models/star.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package models
```
Comment (Bwko) [inline models/star.go:5]: Copyright?

**Comment by bkcsoft** [comment]
LG_TM once copyright is in place :)

**Comment by andreynering** [comment]
Copyright added

File: models/star.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	sess := x.
+		Join("INNER", "star", "star.repo_id = repository.id").
+		Where("star.uid = ?", u.ID)
+
+	if !private {
+		sess = sess.And("is_private = ?", false)
+	}
+
+	err = sess.
+		Find(&repos)
```
Comment (metalmatze) [inline models/star.go:85]: This really doesn't need to be on a separate line, but nitpicking

**Comment by metalmatze** [comment]
LGTM

**Comment by Bwko** [comment]
LGTM

**Pull request merged**